From 8a3321627bb9215ed40f8f7036a1e86bcaf10ad1 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Sun, 22 Nov 2020 08:00:00 +0000
Subject: [PATCH] quotaops: fix compilation warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When quota is configured using --enable-werror --disable-bsd_behaviour,
the compilation fails with the following diagnostics:

quotaops.c: In function 'getprivs':
quotaops.c:143:1: error: label 'out_err' defined but not used [-Werror=unused-label]

Fixes: 7942290a ("quotaops: Do not leak dquot structures on failure")
Signed-off-by: Dmitry V. Levin <ldv@altlinux.org>
Signed-off-by: Jan Kara <jack@suse.cz>
Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 quotaops.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/quotaops.c b/quotaops.c
index ff4d16e..96086f4 100644
--- a/quotaops.c
+++ b/quotaops.c
@@ -140,7 +140,9 @@ struct dquot *getprivs(qid_t id, struct quota_handle **handles, int ignore_noquo
 			id2name(id, handles[i]->qh_type, name);
 			errstr(_("error while getting quota from %s for %s (id %u): %s\n"),
 				handles[i]->qh_quotadev, name, id, estr);
+#if defined(BSD_BEHAVIOUR)
 out_err:
+#endif
 			freeprivs(qhead);
 			return NULL;
 		}
-- 
2.26.2

